♻️🐛 [Trusted Types] Make loadScript, writeScript, and addAttributesToElement Trusted Types compatible

youssef-attia opened the pull request:
This change updates AMP's loadScript, writeScript, and addAttributesToElement functions to be [Trusted Types](https://developer.mozilla.org/en-US/docs/Web/API/Trusted_Types_API) compatible, partial fix to https://github.com/ampproject/amphtml/issues/37297. The functions were acting as wrappers for some functionality for the setAttribute function and were used extensively throughout the codebase with a variety of different values so it made sense to implement legacy conversions to amend the TT violations.

dependabot[bot] committed:
Bump actions/dependency-review-action from 3.0.4 to 3.0.6 (#39067)

Bumps [actions/dependency-review-action](https://github.com/actions/dependency-review-action) from 3.0.4 to 3.0.6.
- [Release notes](https://github.com/actions/dependency-review-action/releases)
- [Commits](https://github.com/actions/dependency-review-action/compare/f46c48ed6d4f1227fb2d9ea62bf6bcbed315589e...1360a344ccb0ab6e9475edef90ad2f46bf8003b1)

---
updated-dependencies:
- dependency-name: actions/dependency-review-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github/codeql-action from 2.3.4 to 2.3.5 (#39046)

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 2.3.4 to 2.3.5.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/f0e3dfb30302f8a0881bb509b044e0de4f6ef589...0225834cc549ee0ca93cb085b92954821a145866)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Iamlowlo committed:
✨ create Adsviu as ads vendor (#39063)

* feat(AmpAd): create Adsviu ads vendor

* fix(Adsviu): modify adsviuAmp variable to pass tests

* fix(Adsviu): modify prettier requirements

* fix(Adsviu): update copyright year

dependabot[bot] committed:
Bump github/codeql-action from 2.3.5 to 2.3.6 (#39075)

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 2.3.5 to 2.3.6.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/0225834cc549ee0ca93cb085b92954821a145866...83f0fe6c4988d98a455712a27f0255212bba9bd4)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @percy/core in /build-system/tasks/visual-diff (#39078)

Bumps [@percy/core](https://github.com/percy/cli/tree/HEAD/packages/core) from 1.24.2 to 1.25.0.
- [Release notes](https://github.com/percy/cli/releases)
- [Commits](https://github.com/percy/cli/commits/v1.25.0/packages/core)

---
updated-dependencies:
- dependency-name: "@percy/core"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @percy/sdk-utils in /build-system/tasks/visual-diff (#39079)

Bumps [@percy/sdk-utils](https://github.com/percy/cli/tree/HEAD/packages/sdk-utils) from 1.24.2 to 1.25.0.
- [Release notes](https://github.com/percy/cli/releases)
- [Commits](https://github.com/percy/cli/commits/v1.25.0/packages/sdk-utils)

---
updated-dependencies:
- dependency-name: "@percy/sdk-utils"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump jsdom from 16.7.0 to 22.1.0 (#39058)

Bumps [jsdom](https://github.com/jsdom/jsdom) from 16.7.0 to 22.1.0.
- [Release notes](https://github.com/jsdom/jsdom/releases)
- [Changelog](https://github.com/jsdom/jsdom/blob/master/Changelog.md)
- [Commits](https://github.com/jsdom/jsdom/compare/16.7.0...22.1.0)

---
updated-dependencies:
- dependency-name: jsdom
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update build-system devDependencies (#39052)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update subpackage devDependencies (#39073)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update linting devDependencies (major) (#36314)

* 📦 Update linting devDependencies

* Import `ast-utils` from `eslint/lib` via `node_modules`, as it is no longer available via the root package's exports

* Fix for async type

* Explicitly include latest `@types/estree` to fix type checks

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Daniel Rozenberg <rodaniel@google.com>
Co-authored-by: Daniel Rozenberg <rodaniel@amp.dev>

renovate[bot] committed:
📦 Update subpackage devDependencies (#39083)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update dependency eslint to v8.42.0 (#39084)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update dependency inquirer to v9.2.7 (#39091)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update linting devDependencies (#39092)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

erwinmombay committed:
Revert "🏗 Add `process.exit(0)` to task runner when tasks finish (#39071)" (#39090)

This reverts commit cd8b40e1462e4eb104ceaae4bd0b509334c9778e.

powerivq committed:
Change Object to object in type decl (#39095)

bypassing checks to fix the lint issues

powerivq committed:
commit (#39097)

force merge requested by @powerivq

danielrozenberg committed:
Temporary fix for e2e tests getting stuck on CircleCI (#39102)

danielrozenberg committed:
Disable saving Git cache on CircleCI when using Mac OS (#39103)

danielrozenberg committed:
🏗 Bump chromedriver to v113 (#39074)

renovate[bot] committed:
📦 Update dependency eslint-plugin-jsdoc to v46.2.5 (#39105)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update dependency geckodriver to v4.0.3 (#39106)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update validator devDependencies (#38856)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

danielrozenberg committed:
Fix temporary workaround for e2e tests in CircleCI (#39109)

danielrozenberg committed:
🏗 Bump esbuild to 0.16.17 and update code with types (#39094)

* Update esbuild to 0.16.17 and update code with types

* Exclude .json files in esbuild-babel plugin

renovate[bot] committed:
📦 Update dependency eslint-plugin-jsdoc to v46.2.6 (#39112)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update subpackage devDependencies (#39093)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

prabeen04 committed:
✨ PubScale ad network registration (#39051)

* PubScale ad network registration

* resolved  prefetch and preconnect conflict for pubscale ad network

* removed prefetch and preconnect to reduce bundle size

* removed optional params in pubscale example

* optional async loading strategy added

danielrozenberg committed:
Remove dependabot (#39119)

renovate[bot] committed:
📦 Update dependency selenium-webdriver to v4.10.0 (#39115)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update dependency rollup to v3.24.0 (#39124)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

danielrozenberg committed:
Create an empty Visual Diff job from skipped "Module" builds, not "Nomodule" (#39125)

danielrozenberg committed:
Disable flaky test `should restrict fullscreen until user interaction` in amp-consent (#39123)

renovate[bot] committed:
📦 Update peter-evans/create-or-update-comment action to v3.0.2 (#39127)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update build-system devDependencies to v7.0.20 (#39126)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update dependency geckodriver to v4.0.4 (#39128)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

powerivq committed:
Remove rcebulko from check-types.js owner required=true (#39098)

renovate[bot] committed:
📦 Update dependency @types/eslint to v8.40.1 (#39131)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update build-system devDependencies to v7.22.5 (#39130)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update dependency preact-render-to-string to v6.1.0 (#39133)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update actions/checkout action to v3.5.3 (#39134)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update subpackage devDependencies (#39137)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update build-system devDependencies (#39138)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update linting devDependencies to v5.59.11 (#39139)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update dependency @types/eslint to v8.40.2 (#39140)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update dependency puppeteer-core to v20.7.0 (#39141)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

estherkim committed:
Use Object for validator jsdoc (#39136)

* use Object for validator/

* one more thanks allan

* also revert formatting

renovate[bot] committed:
📦 Update dependency puppeteer-core to v20.7.1 (#39143)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

danielrozenberg committed:
Disable broken test `should show the body in boilerplate test` (#39151)

renovate[bot] committed:
📦 Update subpackage devDependencies (#39147)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

ychsieh committed:
✨[AMP Story Audio Sticker] add animations and default styles (#38954)

* Audio sticker skeleton code: clicking on the element would unmute the story and hide the element

* Remove validator test and spaces in css file

* Add dependency

* Update z-index

* Update extensions/amp-story-audio-sticker/0.1/amp-story-audio-sticker.css

Co-authored-by: Matias Szylkowski <mszylkowski@google.com>

* Cleaning up extension and css code

* fix

* Remove unnecessary CSS and update the example.

* Fix z-index

* Remove validator

* Add the sticker container and tap hint

* Fix whitespace

* lint

* String concat in jsx

* Default stickers and switching between different states

* Remove duplicate buildCallback

* Use CSS to execute animation delay instead. Have a new helper function to get default sticker

* Change default size to small

* Fix lint

* Fix unit test

* refactors

* Move amp-img src setting to buildcallback

* Fixes

* Fix

* Update extensions/amp-story-audio-sticker/0.1/amp-story-audio-sticker.js

Co-authored-by: Matias Szylkowski <mszylkowski@google.com>

* lint

* Add animation and styles to sticker

* Rename css variable

* Replace borders with box-shadow.
Use sticker-style instead.
Replace mouse and touch events with platform-agnostic events: pointerdown and pointerup

* Fix unit test and merge with main.

* Replace unnecessary classes and event listeners with CSS selectors.
Remove RGB color checks.

* Allow developers to set the CSS variables directly,
instead of taking it as element attributes and do the checks

* Replace box-shadow with filter drop-shadow
This is to create a  dropshadow that conforms to the shape.

* Ensure transitions work on both scaling down and up.

* minor fixes

* Make color variables global.
Combine two animations into one.

* Remove comment

* Use inset: 0 instead.

* Highlight mute button only when the sticker is clicked

* unit test

* Get system UI in the click handler instead of layout callback to prevent system UI from not built yet.
Remove SYSTEM_UI_IS_BUILT state from store service

* Trivial simplification

* Update extensions/amp-story/1.0/amp-story-system-layer.css

Co-authored-by: Matias Szylkowski <mszylkowski@google.com>

* Update extensions/amp-story-audio-sticker/0.1/amp-story-audio-sticker.js

Co-authored-by: Matias Szylkowski <mszylkowski@google.com>

* Fix lint

---------

Co-authored-by: Matias Szylkowski <mszylkowski@google.com>

fra-cali committed:
Implements Neodata's new vendor config for amp-analytics tag. (#39064)

* Iplements Neodata's new vendor config for amp-analytics tag.
Referred to the "Add Neodata Group vendor for amp analytics #38722" Intent-To-Implement issue.

* Iplements Neodata's new vendor config for amp-analytics tag.
Referred to the "Add Neodata Group vendor for amp analytics #38722" Intent-To-Implement issue.

* Iplements Neodata's new vendor config for amp-analytics tag.
Referred to the "Add Neodata Group vendor for amp analytics #38722" Intent-To-Implement issue.
Fixed some newlines.

* Iplements Neodata's new vendor config for amp-analytics tag.
Referred to the "Add Neodata Group vendor for amp analytics #38722" Intent-To-Implement issue.
Added some properties to vendor-requests.json

* Iplements Neodata's new vendor config for amp-analytics tag.
Referred to the "Add Neodata Group vendor for amp analytics #38722" Intent-To-Implement issue.
Fixed request url in vendor-requests.json

* Iplements Neodata's new vendor config for amp-analytics tag.
Referred to the "Add Neodata Group vendor for amp analytics #38722" Intent-To-Implement issue.
Fixed sidId parameter in neodata.json

* Iplements Neodata's new vendor config for amp-analytics tag.
Referred to the "Add Neodata Group vendor for amp analytics #38722" Intent-To-Implement issue.
Fixed sidId parameter in neodata.json

renovate[bot] committed:
📦 Update dependency semver to v7.5.2 (#39157)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

danielrozenberg committed:
Fix formatting (#39161)

renovate[bot] committed:
📦 Update dependency esbuild to v0.18.4 (#36641)

* 📦 Update dependency esbuild to v0.18.4

* Update build code for esbuild changes

* Switch `karma-esbuild` to `karma-esbuild-up`

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Daniel Rozenberg <rodaniel@amp.dev>
Co-authored-by: Daniel Rozenberg <rodaniel@google.com>

danielrozenberg committed:
🏗 Initialize a Mac OS job early, to cache the Git repository before the Safari test job (#39129)

* Initialize a Mac OS job early, to cache the Git repository before the Safari test job

* Parallelize initilize jobs

* Add git-gc --auto step

renovate[bot] committed:
📦 Update build-system devDependencies to v7.0.21 (#39154)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update dependency puppeteer-core to v20.7.2 (#39162)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

jeesoo-dable committed:
Dable: deliver category options to sendLog (#39171)

zombifier committed:
Add proper support for amp-consent tfcd/tfua signals to amp-ad doubleclick and adsense. (#39165)

erwinmombay committed:
Revert "📦 Update dependency @ampproject/worker-dom to v0.34.0 (#39050)" (#39177)

This reverts commit 1c0436f0dce63c927b7d6588914d639ee50a47cd.

danielrozenberg committed:
Remove the `target` field from the server's tsconfig.json (#39164)

erwinmombay committed:
add new root owners (#39180)

aaronshim commented on 3p/3p.js:
```diff
@@ -85,7 +85,20 @@ export function writeScript(win, url, opt_cb) {
   /** @const {!Element} */
   const s = win.document.createElement('script');
-  s.src = url;
+  if (self.trustedTypes && self.trustedTypes.createPolicy) {
```
In the description, we mention that these are treated as legacy conversions because we can't confirm nor deny that any of these inputs are safe/unsafe, and this callsite is blocking overall TT compatibility.

Can we add that in a comment somewhere here so that future visitors to this site will be able to distinguish it from a reviewed safe exemption, like some of your other PRs are?

aaronshim commented on src/core/dom/index.js:
```diff
@@ -156,7 +156,20 @@ export function insertAtStart(root, element) {
 export function addAttributesToElement(element, attributes) {
   for (const attr in attributes) {
-    element.setAttribute(attr, attributes[attr]);
+    if (attr === 'src' && self.trustedTypes && self.trustedTypes.createPolicy) {
```
Same comment as above, a comment describing this as a legacy conversion please!

aaronshim reviewed (commented):
This is great, thanks for the explanation in the conversation! The one thing that I'm wondering-- do you think it's maybe better to either use a wrapper library like safevalues, or if adding a new dependency is forbidden, writing our own lightweight wrappers to distinguish reviewed vs. legacy exemptions?

estherkim committed:
Refactor root owners (#39189)

enovikov11 committed:
🐛 Add uid generation to metrika analytics template (#39178)

The update adds custom client ID generation and ensures the ID is compatible with the one created by regular metrika script: https://github.com/yandex/metrica-tag/blob/main/src/utils/uid/uid.ts#L51

Otherwise the reports provided by the vendor might not be accurate.

powerivq committed:
Revert "Remove the `target` field from the server's tsconfig.json (#39164)" (#39191)

This reverts commit 4e0df9badfb4f65fc4a6af1cb4e00f58a38e6fcb.

erwinmombay committed:
bump worker dom to 0.35 (#39190)

* bump worker dom to 0.35

0.34 contained an off by 1 bug which is fixed in 0.35

* update package-lock.json

emensc52 committed:
✨ Resolve ATTRIBUTION_REPORTING_STATUS in attributionsrc/src for generated pixels (#39132)

* Resolve ATTRIBUTION_REPORTING_STATUS in attributionsrc/src for generated pixels

* Resolve ATTRIBUTION_REPORTING_STATUS in attributionsrc/src for generated pixels

* Update attributionsrc to attributionSrc

* Update attributionsrc to attributionSrc in tests

* Update attributionsrc to attributionSrc in tests

erwinmombay committed:
add ga4 bug report template (#39196)

erwinmombay committed:
add links for Google Analytics 4 instructions (#39197)

* add links for Google Analytics 4 instructions

* fix bad link

* gnore anchor links

estherkim committed:
Run check-links on main (#39198)

renovate[bot] committed:
📦 Update linting devDependencies (#39167)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update subpackage devDependencies (#39168)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update dependency esbuild to v0.18.10 (#39202)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
📦 Update build-system devDependencies (#39201)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

youssef-attia committed:
Merge branch 'ampproject:main' into tt-legacy-conv

youssef-attia commented:
moved to #8 because of rebase issues
